From ea7e420cf90280a6e4ce74e48b0da14395915ba5 Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Mon, 8 Jul 2019 15:48:08 +0200
Subject: [PATCH 4/5] ceph: fix uninitialized return code

clang points out a -Wsometimed-uninitized bug in the modified
ceph_real_mount() function:

fs/ceph/super.c:850:6: error: variable 'err' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
        if (!fsc->sb->s_root) {
            ^~~~~~~~~~~~~~~~
fs/ceph/super.c:885:9: note: uninitialized use occurs here
        return err;
               ^~~
fs/ceph/super.c:850:2: note: remove the 'if' if its condition is always true
        if (!fsc->sb->s_root) {
        ^~~~~~~~~~~~~~~~~~~~~~
fs/ceph/super.c:843:9: note: initialize the variable 'err' to silence this warning
        int err;
               ^
                = 0

Set it to zero if the condition is false.

Fixes: 108f95bfaa56 ("vfs: Convert ceph to use the new mount API")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
(am from https://lore.kernel.org/lkml/20190708134821.587398-1-arnd@arndb.de/)
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 fs/ceph/super.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/ceph/super.c b/fs/ceph/super.c
index 4406ec7018bb..b5ed031ca6ad 100644
--- a/fs/ceph/super.c
+++ b/fs/ceph/super.c
@@ -876,6 +876,8 @@ static int ceph_real_mount(struct fs_context *fc, struct ceph_fs_client *fsc)
 			goto out;
 		}
 		fsc->sb->s_root = root;
+	} else {
+		err = 0;
 	}
 
 	fc->root = dget(fsc->sb->s_root);
-- 
2.22.0

